### PR TITLE
fix(install): ignore dotfiles except for .ghost-cli in dir check

### LINKED ISSUE #868: Ghost local install doesn't ignore dot files
### Issue Summary

When running `ghost local install` in a directory that contains a required dotfile; Installation fails with the message "Current directory is not empty, Ghost cannot be installed here." There are instance when these dot files are necessary or required by other systems. In my case when using Nodenv, Nodenv requires that a .node-version file be in the directory to map the correct local version of node that you would like to use.

### To Reproduce
Assuming that nodenv and the node version you would like to use is installed. Assuming that the ghost-cli is installed and working correctly. Assuming `CWD` is the selected install directory for your ghost site.

1. nodenv local [node-version-number]
2. ghost install local

> Message: Current directory is not empty, Ghost cannot be installed here.
> Debug Information:
>  OS: Mac OS X, v10.14.2
>  Node Version: v10.13.0
>  Ghost-CLI Version: 1.9.8
>  Environment: production
>  Command: 'ghost install local'
> Try running ghost doctor to check your system for known issues.
> You can always refer to https://docs.ghost.org/docs/troubleshooting#section-cli-errors for troubleshooting.

### Technical details:

* Ghost Version: 1.9.8
* Node Version: 10.13.0
* Browser/OS: n/a
* Database: n/a

### Workaround:
A simple work around is to make the node version global, but this defeats the purpose of having a version manager. Another workaround is to move the dot file one level higher; this causes an issue if you are deploying from a repo that use the files on a remote server.

It would be nice to have the ability via the command line to ignore dot files without removing them.

### PR DESCRIPTION
**Pull request by acburdine**
closes #868
- add dir-is-empty util that checks empty directories correctly

TODO:
- [x] fix install cmd tests

### CODE REVIEW

File: lib/utils/dir-is-empty.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const importantDotfiles = ['.ghost-cli'];
+
+module.exports = function dirIsEmpty(dir) {
+    const files = fs.readdirSync(dir);
+
+    if (!files.length) {
+        return true;
+    }
+
+    return every(files, file => file.match(debugLogRegex) || (file.startsWith('.') && !importantDotfiles.includes(file)));
```
Comment (vikaspotluri123) [inline lib/utils/dir-is-empty.js]:
```suggestion
    return files.filter(file => file.match(debugLogRegex) || (file.startsWith('.') && !importantDotfiles.includes(file));
```

Comment (acburdine) [reply lib/utils/dir-is-empty.js]: realized actually that node 6 implements Array.every natively, meaning we don't even need the lodash dep here.

Also, `every` is a better choice for the logic here imo - it returns a boolean consistently and will exit early if one file is found that does not match the condition.

Comment (vikaspotluri123) [reply lib/utils/dir-is-empty.js]: Whoops, brain fart! I was thinking we're returning a file list 🤦‍♂️

Comment (acburdine) [reply lib/utils/dir-is-empty.js]: done 😄

**Pull request merged**